lib/writers/json-writer.js

```diff
@@ -0,0 +1,407 @@
+var _ = require("underscore");
+var get = require('lodash').get;
+var isEmpty = require('lodash').isEmpty;
+var getReadableElement = require('../json/utils').getReadableElement;
+var TEXT_STYLE_TAGS = ['bold', 'italic', 'underlined'];
+var DEFERRED_TYPES = ['th', 'forceWrite'];
+
+exports.writer = writer;
+
+function writer() {
+    return simpleWriter();
+}
+
+function simpleWriter() {
+    function element(node, assetsArray, parentEl) {
```

Comment:
Is there a way to maybe split this out into individual functions rather than all the if statements. It's quite hard to follow.

Reply:
Yes' it's in my todo list already. Planning to break it apart after I get most of the use-case working already.